xray/core/dataset.py

```diff
@@ -1028,6 +1029,76 @@ def sel(self, method=None, **indexers):
         return self.isel(**indexing.remap_label_indexers(self, indexers,
                                                          method=method))
 
+    def isel_points(self, dim='points', **indexers):
+        """Returns a new dataset with each array indexed pointwise along the
+        specified dimension(s).
+
+        This method selects pointwise values from each array and is akin to
+        the NumPy indexing behavior of `arr[[0, 1], [0, 1]]`, except this
+        method does not require knowing the order of each array's dimensions.
+
+        Parameters
+        ----------
+        dim : str or DataArray or pandas.Index, optinal
+            Name of the dimension to concatenate along. This can either be a
+            new dimension name, in which case it is added along axis=0, or an
```

Comment:
Existing dimension names are not valid choices for sel_points, I think. Actually, that's probably an edge case worth writing a test for.